### PR TITLE
[MOB-43] DELETE Remove event endpoint

### PR DESCRIPTION
**Pull request by JM-Abrodos**
## Pull Request Checklist

- [x] **Title**
- [x] **[MOB US' Link](https://mobilemakers.atlassian.net/browse/MOB-43)**
- [ ] **Link PR to US**
- [x] **Add Tests**
- [ ] **Add ENV Variables and Validations**
- [ ] **Documentation Updates**
- [x] **Swagger Updates**

## Testing Guide
**Example** curls:
- Delete event
```
curl --location --request DELETE 'http://localhost:3000/v1/events' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1dWlkIjoiNjY4ZmE1NmQ0MmZmMzljYmE2NTkxNGJlIiwiaWF0IjoxNzIyOTMwMjQzLCJleHAiOjE3Mjg5MzAyNDN9.ZfJIwQA9InbfuoW2fok1QAWqlndO41vdODqkkfG_1nc' \
--data '{
    "eventId": "667031ff634634f43e1b192f"
}'
```
- Login
```
curl --location 'http://localhost:3000/v1/account/login' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "ozounalocal10@yopmail.com",
    "password": "SuperSecret1!"
}'
```
- Join event
```
curl --location 'http://localhost:3000/v1/account/events' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1dWlkIjoiNjY4ZmE1NmQ0MmZmMzljYmE2NTkxNGJlIiwiaWF0IjoxNzIyOTMwMjQzLCJleHAiOjE3Mjg5MzAyNDN9.ZfJIwQA9InbfuoW2fok1QAWqlndO41vdODqkkfG_1nc' \
--header 'Content-Type: application/json' \
--data '{
    "eventId": "667031ff634634f43e1b192f"
}'
```

### CODE REVIEW

File: src/infrastructure/http/controllers/event.controller.ts [hunk 1]
```diff
@@ -81,4 +89,38 @@ export class EventController {
...
+  @UseGuards(AuthGuard)
+  @HttpCode(204)
+  @SwaggerDEleteEvents()
+  async deleteEvent(
+    @Headers('Authorization') accessToken: string,
+    @Body() joinEventDto: EventIdDto,
+    @RequestId() requestId: string,
+  ) {
+    try {
+      // TODO change EventIdDto
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/event.controller.ts:103]: Delete this comment.

File: src/infrastructure/http/controllers/event.controller.ts [hunk 2]
```diff
@@ -81,4 +89,38 @@ export class EventController {
...
+        `[EventController] [deleteEvent] - x-request-id:${requestId}`,
+      );
+
+      await this.deleteEventService.delete(
+        accessToken,
+        joinEventDto.eventId,
+        requestId,
+      );
+
+      return {};
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/event.controller.ts:114]: You don't need to add this return, in fact even if you put an empty object here it is not returned like this, you can delete it.

File: src/infrastructure/http/controllers/event.controller.ts [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
 import {
   CreateEventService,
   GetEventListService,
+  DeleteEventService,
 } from '../../../application/handler/event';
 import { HttpExceptionFilter } from '../filters';
 import {
   SwaggerCreateEvent,
   SwaggerGetEvents,
+  SwaggerDEleteEvents,
```
Comment (LuciaAccenture) [inline src/infrastructure/http/controllers/event.controller.ts:25]: Correct name of this import to `SwaggerDeleteEvents`

File: src/infrastructure/http/swagger/responses/event-responses.ts [hunk 4]
```diff
@@ -86,3 +88,85 @@ export const GET_EVENTS_RESPONSES: { [key: string]: ApiResponseOptions } = {
     },
   },
 };
+
+export const DELETE_EVENT_RESPONSES: { [key: string]: ApiResponseOptions } = {
+  NO_CONTENT: {
+    description: 'Event data has been succesfully delete',
+  },
+  BAD_REQUEST_ERROR: {
```
Comment (LuciaAccenture) [inline src/infrastructure/http/swagger/responses/event-responses.ts:96]: Here you can use the existing constant of `BAD_REQUEST_ERROR_ERROR_COMMON_PROPERTIES` and then you only have to change the path and the error (as it is done in the `CREATE_EVENT_RESPONSES`).
If you do that you can change in `BAD_REQUEST_ERROR_ERROR_COMMON_PROPERTIES` and in `INTERNAL_ERROR_ERROR_COMMON_PROPERTIES` the status example with the HttpStatus.

File: src/application/handler/event/delete-event.service.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    private readonly eventRepository: EventRepository,
+    private readonly accountRepository: AccountRepository,
+    private readonly tokensRepository: TokensRepository,
+    private readonly eventParticipantsRepository: EventParticipantsRepository,
+  ) {}
+
+  async delete(accessToken: string, eventId: string, requestId?: string) {
+    try {
+      this.logger.info(
+        `[DeleteEventService] [delete] - x-request-id:${requestId}`,
```
Comment (LuciaAccenture) [inline src/application/handler/event/delete-event.service.ts:29]: You could add here `eventId: ${eventId}`

File: src/domain/errors/not-acceptable-event-error.ts [hunk 6]
```diff
@@ -1,7 +1,13 @@
+import { HttpStatus } from '@nestjs/common';
 import { ApiError } from './api-error';
 
 export class NotAccepteableEventApiError extends ApiError {
```
Comment (LuciaAccenture) [inline src/domain/errors/not-acceptable-event-error.ts:4]: Misspelled name, change to `NotAcceptableEventApiError`

File: test/jest-component/features/steps/events/delete-events-steps.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+const PATH = '/v1/events';
+
+const METHOD = 'delete';
+
+const HEADERS = {
+  'Content-Type': 'application/json',
+  Authorization: 'Bearer accessToken',
+};
+
+const BODY = { ...JOIN_EVENT_MOCK };
```
Comment (LuciaAccenture) [inline test/jest-component/features/steps/events/delete-events-steps.ts:27]: Maybe it's better to rename the mock the same as the dto and put `EVENT_ID_MOCK`

File: src/infrastructure/http/swagger/responses/event-responses.ts (hunk 4 above)
Comment (JM-Abrodos) [reply src/infrastructure/http/swagger/responses/event-responses.ts:71]: I refactored common properties for badRequest and internalServer errors responses in a common file to use them in account and event bundle context response. I also created a common properties for unhautorized response related to invalid token data, another scenario to handle in a lot of flows

**Comment by JM-Abrodos** [comment]
@LuciaAccenture Changes implemented 🙇

**Review by LuciaAccenture — APPROVED** [review]

**Pull request merged**